### DISCUSSION TITLE
Anyone Can help with SU2 hotspot identification (profiling)...?

### DISCUSSION THREAD

**Discussion by psgmech** [su2code/SU2#1393] [Q&A]
I would like to give a brief intro and the technical difficulty of my work.. Anyone interested to talk further..

**Comment by TobiKattmann** [comment]
Maybe @pcarruscag did sth similiar in his PhD thesis, back in the day. But he'll most likely be at the developers meeting this afternoon as well. 

I sent @psgmech to this place, coming from su2code/Testcases/issues/77 , there are a few more infos

@psgmech to make it easier for anyone to understand and help it is beneficial to just go ahead and present your specifc ideas/problem at hand. Like this you it is much easier to get into some discussion. And maybe someone in the future then sees your approach and can learn from that. I personally do not know too much about profiling and I am definitely interested in that (although I cannot add too much to that)

**Comment by pcarruscag** [comment]
Hi,
#716 and https://www.researchgate.net/publication/348237752_Sustainable_High-Performance_Optimizations_in_SU2

**Reply by TobiKattmann** [reply]
Maybe another interesting reference is [this paper from Thomas Economon](https://www.researchgate.net/publication/294424718_Performance_optimizations_for_scalable_implicit_RANS_calculations_with_SU2) which maybe is  sth like a followup to your paper pedro

**Comment by psgmech** [comment]
Thank you for the reference journal.

We are looking to integrate Su2 C++ code with CUDA C++ (GPU Code).

I will check this article and update you. Thank you so much

**Reply by pcarruscag** [reply]
Ah nice. And so you are looking to start with the more intensive routines and keep some stuff in CPU?
Anyway, looking forward to hear about this if you can join the dev meeting.
IDK if you know this but the SU2 Foundation was offered help by NVIDIA (the expert advice kind, not the "send-us-some-A100's" kind :smile:) if you / your group would be interested in that we should talk.

**Comment by psgmech** [comment]
I will try to join today's devmeeting to discuss the same.

I am basically of computer science background. But my project is to
optimize the SU2 intensive computation codes using GPU.

I learned GPU programming, and could work on standalone c/c++ programs with
CUDA C.

Now, My exact difficulty here is... Unable to find intensive parts of
SU2... How to know the computation intensive parts of SU2 since it has many
modules with many solver c++ files.

Kindly give your guidance on this.

**Comment by psgmech** [comment]
Thanks, I'll check it out.

**Comment by psgmech** [comment]
inputs from yesterday's meeting:

1. Numerics module handles the computational part of SU2
2. Pre-processing and post-processing handled by Integration module
3. Perf  - Linux profiler

I will work on it and update you in the next meeting.
